hflags-0.4.2 does not compile

Trying to hflags with GHC 8.2.1, I get the following error:

~~~
Preprocessing library for hflags-0.4.2..
Building library for hflags-0.4.2..
[1 of 1] Compiling HFlags           ( HFlags.hs, dist/build/HFlags.o )

HFlags.hs:190:81: error:
    • Couldn't match type ‘Q Cxt’ with ‘[DerivClauseQ]’
      Expected type: [DerivClauseQ]
        Actual type: CxtQ
    • In the sixth argument of ‘dataD’, namely ‘(cxt [])’
      In a stmt of a 'do' block:
        dataDec <- dataD
                     (cxt []) dataName [] Nothing [normalC dataConstrName []] (cxt [])
      In the expression:
        do (name, short) <- if | length name' == 0
                                   -> fail "Flag's without names are not supported."
                               | length name' == 1 -> return (name', Just $ head name')
                               | length name' == 2 -> return (name', Nothing)
                               | name' !! 1 == ':' -> return (drop 2 name', Just $ head name')
                               | otherwise -> return (name', Nothing)
           defE <- defQ
           flagType <- case defE of
                         SigE _ flagType -> return $ return flagType
                         _ -> fail
                                "Default value for defineCustomFlag has to be an explicitly typed expression, like (12 :: Int)"
           moduleName <- fmap loc_module location
           ....
    |
190 |      dataDec <- dataD (cxt []) dataName [] Nothing [normalC dataConstrName []] (cxt [])
    |                                                                                 ^^^^^^
~~~~

--- a/HFlags.hs
+++ b/HFlags.hs
@@ -186,7 +186,9 @@ defineCustomFlag name' defQ argHelp readQ showQ description =
      let dataName = mkName $ "HFlag_" ++ name
      let dataConstrName = mkName $ "HFlagC_" ++ name
      -- Note: support for splicing inside [d| |] would make all this a lot nicer
-#if MIN_VERSION_template_haskell(2,11,0)
+#if MIN_VERSION_template_haskell(2,12,0)
+     dataDec <- dataD (cxt []) dataName [] Nothing [normalC dataConstrName []]      []
+#elif MIN_VERSION_template_haskell(2,11,0)
      dataDec <- dataD (cxt []) dataName [] Nothing [normalC dataConstrName []] (cxt [])
 #else
      dataDec <- dataD (cxt []) dataName []         [normalC dataConstrName []]      []

